rasterio/rio/main.py

```diff
@@ -75,3 +74,5 @@ def main_group(ctx, verbose, quiet, aws_profile, gdal_version):
     ctx.obj = {}
     ctx.obj['verbosity'] = verbosity
     ctx.obj['aws_profile'] = aws_profile
+    ctx.obj['env'] = rasterio.Env(CPL_DEBUG=(verbosity > 2),
+                                  profile_name=aws_profile)
```

Comment:
I like the idea of instantiating the Env here, making it available to all commands (including plugins).

It opens up a question: do we want to allow a generic `--configuration-option` or similar to pass arbitrary GDAL config options into the environment from the command line?